wgengine/filter/tailcfg.go

```diff
@@ -28,7 +28,14 @@ func MatchesFromFilterRules(pf []tailcfg.FilterRule) ([]Match, error) {
 	var erracc error
 
 	for _, r := range pf {
-		m := Match{}
+		// Profiling determined that this function was spending a lot
+		// of time in runtime.growslice. As such, we attempt to
+		// pre-allocate some slices. Multipliers were chosen arbitrarily.
```

Comment:
arbitrarily means it could actually be worse, though?

Reply:
growslice is a different codepath to preallocation i think? so we are amortizing multiple allocation & copies to one or at least less.